benchmark/single-source/DictionaryCompactMapValues.swift

```diff
@@ -1,76 +1,35 @@
-// Dictionary compact map values benchmark
+//===--- DictionaryCompactMapValues.swift ---------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2018 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
 import TestsUtils
 
+let size = 100
+let oddNumbers = stride(from: 1, to: size, by: 2)
+let smallOddNumMap: [Int: Int?] =
+  Dictionary(uniqueKeysWithValues: zip(oddNumbers, oddNumbers))
+let compactOddNums: [Int: Int] =
+  Dictionary(uniqueKeysWithValues: zip(oddNumbers, oddNumbers))
+let oddStringMap: [Int: String] = Dictionary(uniqueKeysWithValues:
+  (1...size).lazy.map { ($0, $0 % 2 == 0 ? "dummy" : "\($0)") })
+
+let t: [BenchmarkCategory] = [.validation, .api, .Dictionary]
+
 public let DictionaryCompactMapValues = [
   BenchmarkInfo(name: "DictionaryCompactMapValuesOfNilValue",
-    runFunction: run_DictionaryCompactMapValuesOfNilValue,
```

Comment:
you should add a setUpFunction: { blackHole(inputvariable) }
to force computing the input variable before the benchmark function.

Reply:
Given that this affects only first sample, I thought it is too much boiler plate… but I'd rather get this over with so … done.